File: medicare-frontend/src/types/kpi.ts
export interface KpiSnapshotOut {
  total_consultations: number;
  pending_consultations: number;
  active_subscriptions: number;
  pending_approvals: number;
  total_reviews: number;
  avg_compliance_score: number;
  conversion_rate: number;
  churn_risk_count: number;
  generated_at: string;
}

export interface TrendPoint {
  date: string;
  count: number;
}

export interface VerticalDistribution {
  vertical: string;
  count: number;
  percentage: number;
}

export interface AgentActivitySummary {
  agent_id: string;
  runs_today: number;
  success_rate: number;
}

File: medicare-backend/app/schemas/kpi.py
from datetime import datetime
from pydantic import BaseModel


class KpiSnapshotOut(BaseModel):
    total_consultations: int
    pending_consultations: int
    active_subscriptions: int
    pending_approvals: int
    total_reviews: int
    avg_compliance_score: float
    conversion_rate: float
    churn_risk_count: int
    generated_at: datetime


class TrendPoint(BaseModel):
    date: str
    count: int


class VerticalDistribution(BaseModel):
    vertical: str
    count: int
    percentage: float


class AgentActivitySummary(BaseModel):
    agent_id: str
    runs_today: int
    success_rate: float
